packages/wallet/src/components/kyc/VerifyIdentityForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button } from '@/ui/Button'
+import { FieldError } from '@/ui/forms/FieldError'
+import { FileUpload } from '@/ui/forms/FileUpload'
+import { Form, useZodForm } from '@/ui/forms/Form'
+import { z } from 'zod'
+
+// mock data, the list will come from Rapyd
+const idTypes = [
+  {
+    id: '1',
+    type: 'id',
+    label: 'ID'
+  },
+  {
+    id: '2',
+    type: 'passport',
+    label: 'Passport'
+  },
+  {
+    id: '3',
+    type: 'driverLicence',
+    label: "Driver's licence"
+  }
+]
+
+const verifyIdentitySchema = z
+  .object({
+    idType: z.string({ invalid_type_error: 'Please select an ID Type' }),
+    frontSideID: z.any(),
+    selfie: z.any()
+  })
+  .superRefine(({ selfie, frontSideID }, ctx) => {
+    if (selfie[0] === undefined) {
+      ctx.addIssue({
+        code: 'custom',
+        message: 'Please upload selfie',
+        path: ['selfie']
+      })
+    } else if (frontSideID[0] === undefined) {
+      ctx.addIssue({
+        code: 'custom',
+        message: 'Please upload front of ID document',
+        path: ['frontSideID']
+      })
+    }
+  })
```

Comment:
```suggestion
const verifyIdentitySchema = z
  .object({
    idType: z.string({ invalid_type_error: 'Please select an ID Type' }),
    frontSideID: z
      .custom<FileList>()
      .refine(
        (frontSideID) => frontSideID?.length === 1,
        'Front side of ID is required.'
      )
      .refine(
        (frontSideID) => frontSideID?.length < 2,
        'You can only select one image.'
      ),
    selfie: z
      .custom<FileList>()
      .refine((selfie) => selfie?.length === 1, 'Selfie is required.')
      .refine((selfie) => selfie?.length < 2, 'You can only select one image.')
  })
```

Reply:
Nice, it works this way, the only thing is that there is always one file uploaded even if the user tries to adds more at once, or if they upload 2 one after the other, the length will always be 1. I didn't add multiple in the file upload probably that's why.